### DISCUSSION TITLE
Build fails

### DISCUSSION THREAD

**Discussion by timo-r3** [doudar/SmartSpin2k#201] [Q&A] [answered]
I tried to build the project but looks like linker cannot find crt1-sim.o and _vectors.io. I'm using Windows10, VSCode with PlatformIO extension (as in readme). I tried both develop and master branch. I can find few similar errors with google, but don't know how to fix. Any idea what could be wrong?

```
Linking .pio\build\esp32doit\firmware.elf
   /.platformio/packages/toolchain-xtensa32/bin/../lib/gcc/xtensa-esp32-elf/5.2.0/../../../../xtensa-esp32-elf/bin/ld.exe: cannot find crt1-sim.o: No such file or directory
   /.platformio/packages/toolchain-xtensa32/bin/../lib/gcc/xtensa-esp32-elf/5.2.0/../../../../xtensa-esp32-elf/bin/ld.exe: cannot find _vectors.o: No such file or directory
collect2.exe: error: ld returned 1 exit status
*** [.pio\build\esp32doit\firmware.elf] Error 1
=============================================================================== [FAILED] Took 55.04 seconds ===============================================================================

Environment    Status    Duration
-------------  --------  ------------
esp32doit      FAILED    00:00:55.039
```

**Comment by posssson** [comment, marked as answer]
Find in internet remove this line in the .ini :  platform_packages = framework-arduinoespressif32@https://github.com/espressif/arduino-esp32.git
Works for me :)

**Reply by timo-r3** [reply]
Thanks, it builds now, I will test it later

**Reply by kadaan** [reply]
Alternatively, change: 
```
platform_packages = framework-arduinoespressif32@https://github.com/espressif/arduino-esp32.git#master
```
to:
```
platform_packages = framework-arduinoespressif32@https://github.com/espressif/arduino-esp32.git#master
	            toolchain-xtensa32@~2.80400.0
```

**Reply by kadaan** [reply]
Or pin framework-arduinoespressif32 to the latest 1.x release: 1.0.6:
```
platform_packages = framework-arduinoespressif32@https://github.com/espressif/arduino-esp32.git#1.0.6
```

**Reply by doudar** [reply]
fixed by #216

**Comment by doudar** [comment]
> Find in internet remove this line in the .ini :  platform_packages = framework-arduinoespressif32@https://github.com/espressif/arduino-esp32.git
> 
> Works for me :)

@kadaan , I haven't seen this issue. Any ideas?
